ruby.asciidoc

```diff
@@ -0,0 +1,565 @@
+
+[source,ruby]
+----
+class Greeter
+  def greet
+    puts "Hello World!"
+  end
+end
+
+10.times.map do
+  Thread.new do
+    Greeter.new.greet
+  end
+end.each(&:join)
+----
+
+Here’s a rough port to Pony:
+
+[source,ponylang]
+----
+use "collections"
+
+actor Main
+  new create(env: Env) =>
+    for i in Range[U8](1, 10) do
+      Greeter(env.out).greet()
+    end
+
+actor Greeter
+  let _out: StdStream
+
+  new create(out: StdStream) =>
+    _out = out
+
+  be greet() =>
+    _out.print("Hello world!")
+----
+
+That’s it. Note the stuff that’s similar to Ruby:
+
+* Variables are in `snake_case`
+* We have 'blocks' that use `do` and `end`
+* Variables, while statically typed, have inference, so we don’t need to
+declare types
+
+Here’s some stuff that’s different:
+
+* slightly different syntax, `be` rather than `def` which is an
+asynchronous function which we'll come to later
+
+[[adding-a-little-class]]
+Adding a Little Class
+~~~~~~~~~~~~~~~~~~~~~
+
+In this next example, let’s look at a program with a couple of classes
+that have methods, variables and loops.
+
+Here is the Ruby version:
+
+[source,ruby]
+----
+$ cat measurements/rectangle.rb
+
+class Rectangle
+  attr_reader :width, :height
+
+  def initialize(width, height)
+    @width = width
+    @height = height
+  end
+
+  def dimensions
+    width + " " + height
+  end
+
+  def circumference
+    2 * (width + height)
+  end
+
+  def area
+    width * height
+  end
+end
+
+(1...5).each do |i|
+  r = Rectangle.new(i, i + 2)
+
+  puts "Width and height: #{r.dimensions}"
+  puts "Circumference: #{r.circumference}"
+  puts "Area: #{r.area}"
+  puts ""
+end
+----
+
+In this example we have a `Rectangle` class that encapsualtes a width
+and height, and can compute a couple of attributes of the rectangle.
+
+Our program uses the Rectangle class to print these attributes for a
+sequence of ever larger rectangles.
+
+To run the program, execute these commands in the measurements
+directory:
+
+[source,shell]
+----
+$ ruby rectangle.rb
+----
+
+And the Pony version:
+
+[source,ponylang]
+----
+use "collections"
+
+actor Main
+  new create(env: Env) =>
+    for i in Range[I32](1, 5) do
+        let r = Rectangle(i, i + 2)
+        env.out.print("Width and height: " + r.dimensions())
+        env.out.print("Circumference: " + r.circumference().string())
+        env.out.print("Area: " + r.area().string())
+        env.out.print("")
+    end
+
+class Rectangle
+  let _width: I32
+  let _height: I32
+
+  new create(width: I32, height: I32) =>
+    _width = width
+    _height = height
+
+  fun dimensions(): String => _width.string() + " " + _height.string()
+
+  fun circumference(): I32 => 2 * (_width + _height)
+
+  fun area(): I32 => _width * _height
+----
+
+In the Pony version, the `Rectangle` class structure is essentially the
+same as the Ruby version. The `let` keyword introduces fields or local
+variables that must be initialized exactly once. Visibility in Pony is
```

Comment:
"initialized" could be a loaded/misleading term for Ruby folks, who use it to refer to their constructors.  "assigned" is probably a better word choice here.